finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/service/H2Classifier.scala

```diff
@@ -0,0 +1,23 @@
+package com.twitter.finagle.buoyant.h2.service
+
+import com.twitter.finagle.service.ResponseClass
+
+/**
+ * H2StreamClassifier is similar to Finagle's ResponseClassifier type, but can also classify based
+ * on the final Frame of the Response stream.  It is H2 specific.  When a Response future
+ * completes, the responseClassifier PartialFunction can be checked.  If responseClassifier is
+ * defined then the response can be immediately classified, just as with Finagle's
+ * ResponseClassifier.  If responseClassifier is not defined at that input, we must wait until the
+ * final Frame of the response stream has been read and then use the streamClassifier
+ * PartialFunction to classify the response.
+ */
+trait H2Classifier { self =>
+  def streamClassifier: PartialFunction[H2ReqRepFrame, ResponseClass] = PartialFunction.empty
+  def responseClassifier: PartialFunction[H2ReqRep, ResponseClass] = PartialFunction.empty
```

Comment:
This lets us be more specific with the types and I think we want to explicitly prevent ourselves from accidentally using any of the finagle classification machinery.  There are semantic differences that are subtle.  For example, finagle might classify a 200 response as successful whereas we want the responseClassifier for a 200 response to be undefined so that we can wait and see if the response stream fails.